python/cortex.py

```
@@ -27,16 +29,353 @@
+        elif req_id == MENTAL_COMMAND_TRAINING_THRESHOLD:
+            self.emit('mc_training_threshold_done', data=result_dic)
+        elif req_id == MENTAL_COMMAND_BRAIN_MAP_ID:
+            self.emit('mc_brainmap_done', data=result_dic)
+        elif req_id == SENSITIVITY_REQUEST_ID:
+            self.emit('mc_action_sensitivity_done', data=result_dic)
+        elif req_id == CREATE_RECORD_REQUEST_ID:
+            self.record_id = result_dic['record']['uuid']
+            self.emit('create_record_done', data=result_dic['record'])
+        elif req_id == STOP_RECORD_REQUEST_ID:
+            self.emit('stop_record_done', data=result_dic['record'])
+        elif req_id == EXPORT_RECORD_ID:
+            # handle data lable
+            success_export = []
+            for record in result_dic['success']:
+                record_id = record['recordId']
+                success_export.append(record_id)
+
+            for record in result_dic['failure']:
+                record_id = record['recordId']
+                failure_msg = record['message']
+                print('export_record resp failure cases: '+ record_id + ":" + failure_msg)
+
+            self.emit('export_record_done', data=success_export)
+        elif req_id == INJECT_MARKER_REQUEST_ID:
+            self.emit('inject_marker_done', data=result_dic['marker'])
+        elif req_id == INJECT_MARKER_REQUEST_ID:
+            self.emit('update_marker_done', data=result_dic['marker'])
+        else:
+            print('No handling for response of request ' + req_id)
+
+    def handle_error(self, recv_dic):
+        req_id = recv_dic['id']
+        error_dic = recv_dic['error']
+        error_code = error_dic['code']
+
+        if error_code == ERR_PROFILE_ACCESS_DENIED:
+            #disconnect headset
+            error_message = error_dic['message']
+            warnings.warn(error_message)
+
+            self.disconnect_headset()
+        else:
+            print(recv_dic)
```

Why there's no disconnect_headset call for the other types of error? @tungntEmotiv

---

Print the error to the console would not allow end-user to run their custom error handler, better to raise an explicit exception here with error result

---

I will emit error to main script to handle. Almost cases , will raise a exception to terminate the program and do not need disconnect headset because it might impact to other applications.  Only the case errorcode=-32046, we need to call disconnect to force unload profile for next use.

---

I have updated rework , @thamha-emotiv , @NguyenDuyToan , a @tanpv , Please help to review again. Thanks